src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.Grammar.Test/Directory.Build.props

```diff
@@ -1,11 +1,13 @@
 <Project>
-<PropertyGroup>
-  <!--
-    The Oniguruma package for regex testing has its own Arcade & SDK independent build mechanisms that automatically look for hierarchical Directory.Build.props files.
-    Therefore, we have this file here in order to suppress the usage of the global SDK and arcade build system to enable yarn install to pass.
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory).., Directory.Build.props))\Directory.Build.props" />
```

Comment:
While digging into this I found [GetPathOfFileAbove](https://docs.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2022#msbuild-getpathoffileabove) which seems like a cleaner way of doing this path calculation (in fact, it's even mentioned as a more concise alternative in the docs).
Happy to submit a PR updating all usages of GetDirectoryNameOfFileAbove to GetPathOfFileAbove if it would be merged :).